pkg/webhooks/workload_webhook.go

```diff
@@ -347,29 +347,18 @@ func ValidateWorkloadUpdate(newObj, oldObj *kueue.Workload) field.ErrorList {
 	allErrs = append(allErrs, ValidateWorkload(newObj)...)
 
 	if workload.HasQuotaReservation(oldObj) {
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, specPath.Child("podSets"))...)
```

Comment:
Why remove these?

Reply:
(if you need to change any behavior that is specific to the feature, ensure that it is behind the feature gate, otherwise you will break functionality for other jobs)

Reply:
Sg added feature gate check